### PR TITLE
Added a rule to verify correct Json.Net reference

### PR DESCRIPTION
**Pull request by ajaybhargavb**
@pranavkm / @Eilon

### CODE REVIEW

File: src/NuGetPackageVerifier/Rules/AssemblyHasCorrectJsonNetVersionRule.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+                if (jsonDependency != null && !string.Equals(jsonDependency.VersionSpec.ToString(), "8.0.2"))
+                {
+                    yield return PackageIssueFactory.AssemblyHasWrongJsonNetVersion(
+                        package.Id,
+                        dependencySet.TargetFramework.FullName,
+                        jsonDependency.VersionSpec.ToString());
+                }
+            }
+
+            yield break;
```
Comment (pranavkm) [inline src/NuGetPackageVerifier/Rules/AssemblyHasCorrectJsonNetVersionRule.cs]: Shouldn't need this.

File: src/NuGetPackageVerifier/PackageIssueFactory.cs [hunk 2]
```diff
@@ -131,5 +131,11 @@ public static PackageVerifierIssue AssemblyHasNoDocFile(string assemblyPath)
         {
             return new PackageVerifierIssue("DOC_MISSING", assemblyPath, string.Format("The assembly '{0}' doesn't have a corresponding XML document file.", assemblyPath), MyPackageIssueLevel.Warning);
         }
+
+        public static PackageVerifierIssue AssemblyHasWrongJsonNetVersion(string assemblyPath, string targetFramework, string currentVersion)
+        {
+            return new PackageVerifierIssue("WRONG_JSONNET_VERSION", string.Format("{0}; {1}", assemblyPath, targetFramework), string.Format("The assembly '{0}' references the wrong Json.Net version. Current version '{1}'; Expected version '8.0.2'.", assemblyPath, currentVersion),
```
Comment (pranavkm) [inline src/NuGetPackageVerifier/PackageIssueFactory.cs]: string.Format on a separate line?

**Comment by pranavkm** [comment]
:shipit:

File: src/NuGetPackageVerifier/Rules/AssemblyHasCorrectJsonNetVersionRule.cs (hunk 1 above)
Comment (ajaybhargavb) [inline src/NuGetPackageVerifier/Rules/AssemblyHasCorrectJsonNetVersionRule.cs]: Ah yes. Forgot to remove.

File: src/NuGetPackageVerifier/PackageIssueFactory.cs [hunk 3]
```diff
@@ -131,5 +131,14 @@ public static PackageVerifierIssue AssemblyHasNoDocFile(string assemblyPath)
         {
             return new PackageVerifierIssue("DOC_MISSING", assemblyPath, string.Format("The assembly '{0}' doesn't have a corresponding XML document file.", assemblyPath), MyPackageIssueLevel.Warning);
         }
+
+        public static PackageVerifierIssue AssemblyHasWrongJsonNetVersion(string assemblyPath, string targetFramework, string currentVersion)
+        {
+            return new PackageVerifierIssue(
+                "WRONG_JSONNET_VERSION",
+                string.Format("{0}; {1}", assemblyPath, targetFramework),
+                string.Format("The assembly '{0}' references the wrong Json.Net version. Current version '{1}'; Expected version '8.0.2'.", assemblyPath, currentVersion),
```
Comment (Eilon) [inline src/NuGetPackageVerifier/PackageIssueFactory.cs]: James writes it as `Json.NET` so we should follow that casing.

Comment (Eilon) [inline src/NuGetPackageVerifier/PackageIssueFactory.cs]: (Just in the message, not in the method name or anything.)

File: src/NuGetPackageVerifier/Program.cs [hunk 4]
```diff
@@ -78,6 +78,7 @@ public static int Main(string[] args)
                 new RequiredPackageMetadataRule(),
                 new SatellitePackageRule(),
                 new StrictSemanticVersionValidationRule(),
+                new AssemblyHasCorrectJsonNetVersionRule(),
```
Comment (Eilon) [inline src/NuGetPackageVerifier/Program.cs]: These used to be alphabetical :cry: :panda_face:

Comment (ajaybhargavb) [inline src/NuGetPackageVerifier/Program.cs]: Will address these comments as part of https://github.com/aspnet/DnxTools/issues/20. :smile:

Comment (Eilon) [inline src/NuGetPackageVerifier/Program.cs]: No prob!

**Pull request merged**